apps/server/src/modules/federal-state/repo/federal-state.repo.ts

```diff
@@ -1,41 +1,72 @@
 import { Injectable } from '@nestjs/common';
+import { BaseRepo } from '@shared/repo/base.repo';
 import { FederalStateDO } from '../domainobject';
-import { FederalStateEntity } from '../entity';
+import { County, FederalStateEntity } from '../entity';
+import { IFederalStateCreate } from '../interface';
 
 @Injectable()
-export class FederalStateRepo {
-	constructor(private readonly em: EntityManager) {}
-
+export class FederalStateRepo extends BaseRepo<FederalStateEntity> {
 	get entityName(): EntityName<FederalStateEntity> {
 		return FederalStateEntity;
 	}
 
 	async findByName(name: string): Promise<FederalStateDO> {
-		const federalStateEntity = await this.em.findOneOrFail(FederalStateEntity, { name });
-		const federalState = this.mapEntityToDomainObject(federalStateEntity);
+		// const federalStateEntity = await this.findOneOrFail(FederalStateEntity, { name });
+		const federalStateEntity = await this._em.findOneOrFail(FederalStateEntity, { name });
+		const federalState = this.mapFederalStateEntityToDomainObject(federalStateEntity);
 		return federalState;
 	}
 
 	async findAll(): Promise<FederalStateDO[]> {
-		const federalStateEntities = await this.em.find(FederalStateEntity, {});
+		const federalStateEntities = await this._em.find(FederalStateEntity, {});
+		await this._em.populate(federalStateEntities, ['counties']);
 		const federalStates = federalStateEntities.map((federalStateEntity) =>
-			this.mapEntityToDomainObject(federalStateEntity)
+			this.mapFederalStateEntityToDomainObject(federalStateEntity)
 		);
 		return federalStates;
 	}
 
-	mapEntityToDomainObject(entity: FederalStateEntity): FederalStateDO {
+	async deleteFederalState(federalStateId: string): Promise<FederalStateDO> {
+		const federalStateEntity = await this.findById(federalStateId);
+		await this.delete(federalStateEntity);
+		const federalState = this.mapFederalStateEntityToDomainObject(federalStateEntity);
+		return federalState;
+	}
+
+	async createFederalState(federalStateCreate: IFederalStateCreate): Promise<FederalStateDO> {
+		const federalStateEntity = new FederalStateEntity({
+			...federalStateCreate,
+			createdAt: new Date(),
+			updatedAt: new Date(),
+		});
+		await this.save(federalStateEntity);
+		const federalStateDo = this.mapFederalStateEntityToDomainObject(federalStateEntity);
+		return federalStateDo;
+	}
+
+	mapFederalStateEntityToDomainObject(entity: FederalStateEntity): FederalStateDO {
 		const federalStateDo = new FederalStateDO({
 			id: entity.id,
 			name: entity.name,
 			abbreviation: entity.abbreviation,
 			logoUrl: entity.logoUrl,
-			counties: entity.counties,
+			counties: entity.counties
+				? entity.counties.map((county) => this.mapCountyEntityToDomainObject(county))
+				: undefined,
 			createdAt: entity.createdAt,
 			updatedAt: entity.updatedAt,
 		});
 		return federalStateDo;
 	}
+
+	mapCountyEntityToDomainObject(countyEntity: County) {
+		console.log('countyEntity', countyEntity);
+		const county = {
```

Comment:
no return type for the function.... 
no type for county? no factory / class to instantiate it as an object - just a plain object
mapping should not belong here in my opinion